cmd/process-agent/collector.go

```diff
@@ -196,14 +201,19 @@ func (l *Collector) postMessage(checkPath string, m model.MessageBody) {
 	containerCount := getContainerCount(m)
 
 	responses := make(chan postResponse)
-	for _, ep := range l.cfg.APIEndpoints {
+	endpoints := l.cfg.APIEndpoints
```

Comment:
A function like `endpointsForCheck` would be cleaner here (and then we can just pass in the `check` instead of `checkPath` and use both).